cypress/integration/home.test.ts

```diff
@@ -0,0 +1,355 @@
+
+    option.should("contain", `${hits[3].entry.googleBook?.title}`);
+
+    const optionAuthors: string[] = hits[3].entry.googleBook?.authors || [];
+    const authors = `${optionAuthors[0]}, ${optionAuthors[1]}, ${optionAuthors[2]}`;
+    option.should("contain", authors);
+
+    const optionCourses = hits[3].entry.courses || [];
+    option.should(
+      "contain",
+      `${optionCourses[0].dept?.abbreviation} ${optionCourses[0].code}`
+    );
+    option.should(
+      "contain",
+      `${optionCourses[1].dept?.abbreviation} ${optionCourses[1].code}`
+    );
+    option.should(
+      "contain",
+      `${optionCourses[2].dept?.abbreviation} ${optionCourses[2].code}`
+    );
+    option.should(
+      "not.contain",
+      `${optionCourses[3].dept?.abbreviation} ${optionCourses[3].code}`
+    );
+
+    option
+      .findByRole("img", { name: /book image/i })
+      .should("have.attr", "src")
+      .should("not.be.empty");
+  });
+
+  it("should allow a user to view the content of a book with no google books data", () => {
+    cy.visit("/");
+
+    cy.intercept("**/queries?x-algolia-agent=**", {
+      body: ALGOLIA_RESPONSE_5_RESULTS,
+    }).as("algoliaSearch");
+
+    const hits = ALGOLIA_RESPONSE_5_RESULTS.results[0].hits;
+
+    cy.findByRole("searchbox", { name: /find a book or course/i }).type("test");
+
+    const option = cy.findByRole("listbox").findAllByRole("option").eq(4);
+
+    option.should("contain", `${hits[4].entry.name}`);
+
+    const optionCourses = hits[4].entry.courses || [];
+    option.should(
+      "contain",
+      `${optionCourses[0].dept?.abbreviation} ${optionCourses[0].code}`
+    );
+
+    option
+      .findByRole("img", { name: /book image/i })
+      .should("have.attr", "src")
+      .should("not.be.empty");
+  });
+
+  it("should allow a user to search, click on search button, and be directed to search page", () => {
+    cy.visit("/");
+
+    cy.intercept("**/queries?x-algolia-agent=**", {
+      body: ALGOLIA_RESPONSE_5_RESULTS,
+    }).as("algoliaSearch");
+
+    cy.intercept("**/query?x-algolia-agent=**", {
+      body: ALGOLIA_RESPONSE_5_RESULTS.results[0],
+    }).as("algoliaSearch");
+
+    cy.findByRole("searchbox", { name: /find a book or course/i }).type("test");
+
+    cy.findByRole("button", { name: /search/i }).click({ force: true });
+
+    cy.url().should("contain", "https://localhost:4000/search?q=");
+  });
+
+  it("should allow a user to search, click enter, and be directed to search page", () => {
+    cy.visit("/");
+
+    cy.intercept("**/queries?x-algolia-agent=**", {
+      body: ALGOLIA_RESPONSE_5_RESULTS,
+    }).as("algoliaSearch");
+
+    cy.intercept("**/query?x-algolia-agent=**", {
+      body: ALGOLIA_RESPONSE_5_RESULTS.results[0],
+    }).as("algoliaSearch");
+
+    cy.findByRole("searchbox", { name: /find a book or course/i }).type(
+      "test{enter}"
+    );
+
+    cy.url().should("contain", "https://localhost:4000/search?q=");
```

Comment:
Can you get this from one of the environment variables (see `BASE_URL`)? Don't want these to break if the port or host changes.